lib/chef/application/client.rb

```diff
@@ -37,6 +37,12 @@ class Chef::Application::Client < Chef::Application
     :long  => "--config CONFIG",
     :description => "The configuration file to use"
 
+  option :enable_telemetry,
+    :short => "-T",
+    :long  => "--enable-telemetry",
```

Comment:
im implementing the RFC as it is. I dont want the --telmetry log discussion here. CLI only enabled/disable it. use config options to set specific publishing mechanism.